src/magnify/apps/core/utils.py

```diff
@@ -44,49 +50,70 @@ def get_nth_week_number(original_date):
     return nb_weeks
 
 
-def create_token_payload(user, room, is_admin=False):
-    """Create the payload so that it contains each information jitsi requires"""
+def get_publish_sources(room, is_admin: bool):
+    sources = ["camera", "microphone", "screen_share", "screen_share_audio"]
+    if not room.configuration["screenSharingEnabled"]:
+        sources.remove("screen_share")
+        sources.remove("screen_share_audio")
+    return sources
+
+
+def create_video_grants(room: string, is_admin=False, is_temp_room=True):
```

Comment:
We can simplify these calls to the VideoGrants constructor by creating a base list of arguments. Then, depending on your conditions, you can enrich or override this base list with additional arguments.